stl/inc/ranges

```diff
@@ -1435,20 +1436,101 @@ namespace ranges {
+        owning_view& operator=(owning_view&&) = default;
+
+        _NODISCARD constexpr _Rng& base() & noexcept {
+            return _Range;
+        }
+        _NODISCARD constexpr const _Rng& base() const& noexcept {
+            return _Range;
+        }
+        _NODISCARD constexpr _Rng&& base() && noexcept {
+            return _STD move(_Range);
+        }
+        _NODISCARD constexpr const _Rng&& base() const&& noexcept {
+            return _STD move(_Range);
+        }
+
+        _NODISCARD constexpr iterator_t<_Rng> begin() noexcept(noexcept(_RANGES begin(_Range))) /* strengthened */ {
+            return _RANGES begin(_Range);
+        }
+
+        _NODISCARD constexpr sentinel_t<_Rng> end() noexcept(noexcept(_RANGES end(_Range))) /* strengthened */ {
+            return _RANGES end(_Range);
+        }
+
+        _NODISCARD constexpr auto begin() const
+            noexcept(noexcept(_RANGES begin(_Range))) /* strengthened */ requires range<const _Rng> {
+            return _RANGES begin(_Range);
+        }
+
+        _NODISCARD constexpr auto end() const
+            noexcept(noexcept(_RANGES end(_Range))) /* strengthened */ requires range<const _Rng> {
+            return _RANGES end(_Range);
+        }
+
+        _NODISCARD constexpr bool empty() noexcept(noexcept(_RANGES empty(_Range))) /* strengthened */
+            requires _Can_empty<_Rng> {
+            return _RANGES empty(_Range);
+        }
+        _NODISCARD constexpr bool empty() const noexcept(noexcept(_RANGES empty(_Range))) /* strengthened */
+            requires _Can_empty<const _Rng> {
+            return _RANGES empty(_Range);
+        }
+
+        _NODISCARD constexpr auto size() noexcept(noexcept(_RANGES size(_Range))) /* strengthened */
+            requires sized_range<_Rng> {
+            return _RANGES size(_Range);
+        }
+        _NODISCARD constexpr auto size() const noexcept(noexcept(_RANGES size(_Range))) /* strengthened */
+            requires sized_range<const _Rng> {
+            return _RANGES size(_Range);
+        }
+
+        _NODISCARD constexpr auto data() noexcept(noexcept(_RANGES data(_Range))) /* strengthened */
+            requires contiguous_range<_Rng> {
+            return _RANGES data(_Range);
+        }
+        _NODISCARD constexpr auto data() const noexcept(noexcept(_RANGES data(_Range))) /* strengthened */
+            requires contiguous_range<const _Rng> {
+            return _RANGES data(_Range);
+        }
+    };
+
+    template <class _Rng>
+    inline constexpr bool enable_borrowed_range<owning_view<_Rng>> = enable_borrowed_range<_Rng>;
+
     namespace views {
         template <class _Rng>
         concept _Can_ref_view = requires(_Rng&& __r) {
             ref_view{static_cast<_Rng&&>(__r)};
         };
 
         template <class _Rng>
-        concept _Can_subrange = requires(_Rng&& __r) {
-            subrange{static_cast<_Rng&&>(__r)};
+        concept _Pwnable = requires(_Rng&& __r) {

```

Comment:
Should that be `_Ownable` ? 

That said, the typo might be worth keeping

Also can we change this to `_Is_ownable` and use `_Ownable` for the aforementioned helper concept?